### ISSUE TITLE
Match address format restrictions with tor 0.4.5.7

### ISSUE THREAD

**Issue by mutatrum** [lightningnetwork/lnd#5318]
### Background

When moving my LND instance to a new box, tor was upgraded from 0.4.2.7 to  0.4.5.7. With this new version, tor is more strict on the address format. In my case, the `externalip` field had my external ip and port with two `:`

So `externalip=##.###.###.###::9735`.

The result was that LND failed to start, without leaving any trace in the logs. When starting LND from the command line, the only response was:

```
$ lnd
tor general failure
```

The lack of feedback made this very confusing, as with the previous version of TOR, this configuration was running fine. Tor logs only showed:

```
Your application (using socks5 to port 0) gave Tor a malformed hostname: [scrubbed]. Rejecting the connection.
```

By default tor uses safe logging, which made it even less clear what was going on. Disabling safe logging by setting `SafeLogging=0`  in the tor config revealed the actual address, which I could trace back to the `externalip` setting in `lnd.conf`.

### Your environment

* LND 0.12.1 aarch64
* Ubuntu 21.04
* Bitcoind 0.21.1
* Tor 0.4.5.7

Relevant configuration:
```
tor.active=true
tor.v3=true
tor.streamisolation=true
```

### Steps to reproduce

Set two `:`  between hostname and port in `externalip` configuration, running over tor.

### Expected behaviour

Several options:
- Match address validation to tor, and give direct feedback. Don't accept <ipadress>::<port> (with two semicolons). Maybe other malformed addresses exist which are accepted by LND and not by TOR;
- Capture the error from tor (if possible) and report this back to the user;
- Maybe alternative options?

### Actual behaviour

Failure to start with little to no feedback.

**Comment by CoderBoogs** [comment]
Background

I ran a casa node #1 with LND. Due to power failures the node corrupted. I was not able to back up my channels. But I had my seed. I transferred that node to an Umbrel and noticed the wallet had a balance. Shortly after opening up new channels I noticed my node had old channels that are open on 1ml.com but not visible on thunderhub. I proceeded to open channels with the goal of becoming a routing node and noticed I am unable to change my fees. thunder hub and RTL give me an error. On RTL thee error code is 500 and the error message is channel from self node has no policy  and then an API URL 

Your environment

raspberry pi 4
umbrel  (latest version)
LND 0.12.1
Bitcoind 0.21.1


Relevant configuration:

tor.active=true


Steps to reproduce:

Click on thunderhub channels. click on update fee. change the fee from 1 sat to 2 stats and the unexpected error pops up. 

Expected behavior:
Changing of the fees on my channels 

Several options:

I saw a fix to this potential issue 

https://github.com/lightningnetwork/lnd/issues/2911#issuecomment-490754329

but when I reached out to the Umbrel team they suggested:

You would have to fork our LND container with that patch applied and then modify a configuration file to point to your container. It will get removed after each Umbrel update so you'd have to re-apply the change and they recommended setting up a node and recovering my funds there. 

But I have opened many channels already and this will be too expensive to start over with a fresh node.
